src/DI/PhpReflection.php

```diff
@@ -59,13 +59,19 @@ public static function getDeclaringClass(\ReflectionProperty $prop)
 	 */
 	public static function getPropertyType(\ReflectionParameter $prop)
 	{
-		try {
-			return ($ref = $prop->getClass()) ? $ref->getName() : NULL;
-		} catch (\ReflectionException $e) {
-			if (preg_match('#Class (.+) does not exist#', $e->getMessage(), $m)) {
-				return $m[1];
+		if (PHP_VERSION_ID >= 70000) {
+			return (string) $prop->getType();
+		} elseif ($prop->isArray()) {
```

Comment:
Oh, I understood that you want it in a separate PR afterwards (based on the discussion in this thread on outdated diff), but I'll add it now then and we'll just rename the commit to something like "added support for PHP 7 type hints and PHP 5 callable"? I have it prepared in stash, so it doesn't matter to me if it goes in in this PR on next one. :)